table/metadata.go

```diff
@@ -370,26 +928,34 @@ func (c *commonMetadata) validate() error {
 
 func (c *commonMetadata) Version() int { return c.FormatVersion }
 
-type MetadataV1 struct {
-	Schema    iceberg.Schema           `json:"schema"`
+type metadataV1 struct {
+	Schema    *iceberg.Schema          `json:"schema"`
 	Partition []iceberg.PartitionField `json:"partition-spec"`
 
 	commonMetadata
 }
 
-func (m *MetadataV1) Equals(other Metadata) bool {
-	rhs, ok := other.(*MetadataV1)
+func (m *metadataV1) Equals(other Metadata) bool {
+	rhs, ok := other.(*metadataV1)
 	if !ok {
 		return false
 	}
 
-	return m.Schema.Equals(&rhs.Schema) && slices.Equal(m.Partition, rhs.Partition) &&
+	if m == rhs {
+		return true
+	}
+
+	if m == nil || rhs == nil {
+		return false
+	}
```

Comment:
what scenario can this happen in?

Reply:
My apologies, you are right - this should never happen. I somehow missed this one (and the next comment) when I was cleaning up after my nil-checking frenzy.

I have removed the check.